.github/workflows/wheels.yml

```diff
@@ -0,0 +1,126 @@
+# Workflow to build and test wheels
+name: Wheel builder
+
+on:
+  schedule:
+    # Nightly build at 3:42 A.M.
+    - cron: "42 3 */1 * *"
+  push:
+    tags:
+      # Final releases, release candidates and post-releases build
+      - "[0-9]+.[0-9]+.[0-9]+(rc[0-9]+)?(.post[0-9]+)?"
+  pull_request:
+    branches:
+      - master
+      - "[0-9]+.[0-9]+.X"
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Avoid that the wheel builder for a system is cancelled if another fails
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+
+    steps:
+      - name: Checkout scikit-learn
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+    
+      - name: Install dependencies
+        run: |
+          python -m pip install cibuildwheel
+
+      - name: Build and test wheels
+        env:
+          # Build the wheels for Python 3.6 and newer
+          CIBW_SKIP: pp* *p27* *p35*
```

Comment:
Is this building everything not filtered by `CIBW_SKIP` in the [cibuildwheel matrix](https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip) + running the tests?

This may be okay since [github has a 6 hours](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits) runtime for a workflow.

Reply:
> Is this building everything not filtered by `CIBW_SKIP` in the [cibuildwheel matrix](https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip) + running the tests?
> 

That is the idea!

We specify with `CIBW_TEST_REQUIRES` the testing requirements and with `CIBW_TEST_COMMAND` the test command to run. I have directly taken these values from the [original repository](https://github.com/MacPython/scikit-learn-wheels).

> This may be okay since [github has a 6 hours](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits) runtime for a workflow.

According to previous executions, the wheel builder takes (approximately) 2 hours, while the source distribution 20 minutes. So, I think that it is okay.

We could (strongly) reduce the time by using one job for each matrix combination.

WDYT?

Reply:
I think we can start as is, and move to separate jobs if we encounter any issues.